libbeat/outputs/tidb_slow_query/schema.go

```diff
@@ -0,0 +1,191 @@
+	}
+}
+
+const dateFormat = "2006-01-02"
+
+func insertStmt(schema, table string) string {
+	cols := make([]string, 0, len(orderedColumn))
+	args := make([]string, 0, len(orderedColumn))
+	for _, c := range orderedColumn {
+		cols = append(cols, quote(c))
+		args = append(args, "?")
+	}
+	buf := new(bytes.Buffer)
+	buf.WriteString(fmt.Sprintf("INSERT INTO %s.%s (", quote(schema), quote(table)))
+	buf.WriteString(strings.Join(cols, ","))
+	buf.WriteString(") ")
+	buf.WriteString("VALUES (")
+	buf.WriteString(strings.Join(args, ","))
+	buf.WriteString(") ")
+	buf.WriteString(";")
+	return buf.String()
+}
+
+func createTableStmt(schema, table string, lessThanPartitions []time.Time) string {
+	buf := new(bytes.Buffer)
+	buf.WriteString(fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s.%s (", quote(schema), quote(table)))
+	buf.WriteString("`id` bigint(20) unsigned not null AUTO_INCREMENT,")
+	for k, v := range slowQuerySQLType {
+		buf.WriteString(fmt.Sprintf("%s %s,", quote(k), v))
+	}
+	buf.WriteString("PRIMARY KEY (`id`,`Time`)")
+	buf.WriteString(") ")
+	buf.WriteString("PARTITION BY RANGE (FLOOR(UNIX_TIMESTAMP(`Time`))) (")
+	for _, p := range lessThanPartitions {
+		unix := p.Unix()
+		buf.WriteString(fmt.Sprintf("PARTITION %s VALUES LESS THAN (%d),", partName(p), unix))
+	}
+	// delete the last ,
+	if len(lessThanPartitions) > 0 {
+		buf.Truncate(buf.Len() - 1)
+	}
+	buf.WriteString(");")
+	return buf.String()
+}
+
+func calculateLessThanPartitionBoundary(t time.Time, step int) []time.Time {
+	parts := make([]time.Time, 0, step)
+	// ensure start of the day at +8:00 timezone
+	y, m, d := t.In(zone).Date()
+	round := time.Date(y, m, d, 0, 0, 0, 0, zone)
+	for i := 1; i <= step; i++ {
+		parts = append(parts, round.Add(time.Duration(i)*24*time.Hour))
+	}
+	return parts
+}
+
+func getPartitionStmt(schema, table string) string {
+	return fmt.Sprintf("SELECT `partition_name` FROM `information_schema`.`partitions` "+
+		"WHERE table_schema='%s' AND table_name='%s' AND `partition_name` IS NOT NULL order by `partition_name` asc",
+		schema, table)
+}
+
+func creationPartitionStmt(schema, table string, lessThanPartitions []time.Time) string {
+	buf := new(bytes.Buffer)
+
+	buf.WriteString(fmt.Sprintf("ALTER TABLE %s.%s ADD PARTITION (", quote(schema), quote(table)))
+	for _, p := range lessThanPartitions {
+		unix := p.Unix()
+		buf.WriteString(fmt.Sprintf("PARTITION %s VALUES LESS THAN (%d),", partName(p), unix))
+	}
+	// delete the last ,
+	if len(lessThanPartitions) > 0 {
+		buf.Truncate(buf.Len() - 1)
+	}
+	buf.WriteString(");")
+	return buf.String()
+}
+
+func dropPartitionStmt(schema, table string, lessThanPartitions []string) string {
+	buf := new(bytes.Buffer)
+
+	buf.WriteString(fmt.Sprintf("ALTER TABLE %s.%s drop PARTITION ", quote(schema), quote(table)))
+	for _, p := range lessThanPartitions {
+		buf.WriteString(quote(p) + ",")
+	}
+	// delete the last ,
+	if len(lessThanPartitions) > 0 {
+		buf.Truncate(buf.Len() - 1)
```

Comment:
You can implement the `,` truncation by not appending the `,` for the last item (or not prepending the `,` for the first item). There is no need to write and then truncate.